pkg/sql/crdb_internal.go

```diff
@@ -1179,32 +1179,55 @@ CREATE TABLE crdb_internal.jobs (
 	comment: `decoded job metadata from crdb_internal.system_jobs (KV scan)`,
 	indexes: []virtualIndex{{
 		populate: func(ctx context.Context, unwrappedConstraint tree.Datum, p *planner, _ catalog.DatabaseDescriptor, addRow func(...tree.Datum) error) (matched bool, err error) {
-			q := jobsQuery + jobIDFilter
 			targetID := tree.MustBeDInt(unwrappedConstraint)
-			return makeJobsTableRows(ctx, p, addRow, q, targetID)
+			return makeJobsTableRows(ctx, p, addRow, jobIDFilter, targetID)
 		},
 	}, {
 		populate: func(ctx context.Context, unwrappedConstraint tree.Datum, p *planner, _ catalog.DatabaseDescriptor, addRow func(...tree.Datum) error) (matched bool, err error) {
-			q := jobsQuery + jobsStatusFilter
 			targetStatus := tree.MustBeDString(unwrappedConstraint)
-			return makeJobsTableRows(ctx, p, addRow, q, targetStatus)
+			return makeJobsTableRows(ctx, p, addRow, jobsStatusFilter, targetStatus)
 		},
 	}, {
 		populate: func(ctx context.Context, unwrappedConstraint tree.Datum, p *planner, _ catalog.DatabaseDescriptor, addRow func(...tree.Datum) error) (matched bool, err error) {
-			q := jobsQuery + jobsTypeFilter
 			targetType := tree.MustBeDString(unwrappedConstraint)
-			return makeJobsTableRows(ctx, p, addRow, q, targetType)
+			return makeJobsTableRows(ctx, p, addRow, jobsTypeFilter, targetType)
 		},
 	}},
 	populate: func(ctx context.Context, p *planner, db catalog.DatabaseDescriptor, addRow func(...tree.Datum) error) error {
-		_, err := makeJobsTableRows(ctx, p, addRow, jobsQuery)
+		_, err := makeJobsTableRows(ctx, p, addRow, "")
 		return err
 	},
 }
 
+var useOldJobsVTable = settings.RegisterBoolSetting(
+	settings.ApplicationLevel,
+	"sql.jobs.legacy_vtable.enabled",
+	"cause the crdb_internal.jobs vtable to be produced from the legacy payload info records",
+	false, // TODO(dt): flip this once we add permissive auth checks.
```

Comment:
can we make this metamorphic to maintain test coverage of the old vtable? Also i think the todo can be removed.

Reply:
We get it in mixed version. I was planning     on removing the setting outright as soon as the backport lands so there won’t be any coverage to maintain.

Reply:
I think we want to maintain coverage on the 25.1 release branch?

Reply:
we have that via the version gate already though. This isn't a public setting so I'm not sure it seems worth an extra metamorphic just to cover it?

Reply:
resolved in person. bors away!